src/main/java/nextstep/subway/auth/ui/AuthenticationPrincipalArgumentResolver.java

```diff
@@ -26,6 +28,7 @@ public boolean supportsParameter(MethodParameter parameter) {
     @Override
     public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer, NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
         String credentials = AuthorizationExtractor.extract(webRequest.getNativeRequest(HttpServletRequest.class));
-        return authService.findMemberByToken(credentials);
+        boolean isRequired = Objects.requireNonNull(parameter.getParameterAnnotation(AuthenticationPrincipal.class)).required();
```

Comment:
```suggestion
        boolean isRequired = parameter.getParameterAnnotation(AuthenticationPrincipal.class).required();
```
이 부분은 궁금하여 여쭤봅니다. 이런 식으로 해줘도 동작할 것 같은데, Objects.requireNonNull를 쓰신 이유가 궁금합니다..!

Reply:
아 요렇게 한이유는..! 

인텔리제이에서 한번 감싸는게 좋다고 추천을 해주고 있어, 감싸도록 하였습니다..!

![image](https://user-images.githubusercontent.com/43979984/207581107-d2e45317-696b-4732-ba77-bf157f20391d.png)